packages/flutter/lib/src/painting/box_border.dart

```diff
@@ -808,9 +807,9 @@ class BorderDirectional extends BoxBorder {
   void paint(
     Canvas canvas,
     Rect rect, {
-    TextDirection textDirection,
+    TextDirection? textDirection,
```

Comment:
We assert this isn't null. If we have to make it required that's probably better than making it nullable.

Reply:
Or is this because the super class lets it be null?  Is there a big impact to changing the superclass in that case?

Reply:
Same here
https://github.com/flutter/flutter/blob/bdadd7f4e88b35c976bb3160a5c312fa4bbe09c4/packages/flutter/lib/src/painting/borders.dart#L483-L487